lib/ed25519.js

```diff
@@ -0,0 +1,147 @@
+ * @param {Buffer} [options.seedBytes] - Required if no privateKeyBytes.
+ *
+ * @throws {TypeError} Throws if the supplied buffer is not of the right size
+ *  or not a Uint8Array or Buffer.
+ *
+ * @returns {Buffer} DER private key prefix + key bytes.
+*/
+function privateKeyDerEncode({privateKeyBytes, seedBytes}) {
+  if(!(privateKeyBytes || seedBytes)) {
+    throw new TypeError('`privateKeyBytes` or `seedBytes` is required.');
+  }
+  if(!privateKeyBytes && !(seedBytes instanceof Uint8Array &&
+    seedBytes.length === 32)) {
+    throw new TypeError('`seedBytes` must be a 32 byte Buffer.');
+  }
+  if(!seedBytes && !(privateKeyBytes instanceof Uint8Array &&
+    privateKeyBytes.length === 64)) {
+    throw new TypeError('`privateKeyBytes` must be a 64 byte Buffer.');
+  }
+  let p;
+  if(seedBytes) {
+    p = seedBytes;
+  } else {
+    // extract the first 32 bytes of the 64 byte private key representation
+    p = privateKeyBytes.slice(0, 32);
+  }
+  return Buffer.concat([DER_PRIVATE_KEY_PREFIX, Buffer.from(p)]);
+}
+
+/**
+ * Takes a Buffer of public key bytes and encodes it in DER-encoded SubjectPublicKeyInfo (SPKI) format.
+ * Allows Uint8Arrays to be interoperable with node's crypto functions.
+ *
+ * @param {object} options - Options to use.
+ * @param {Uint8Array} options.publicKeyBytes - The keyBytes.
+ *
+ * @throws {TypeError} Throws if the bytes are not Uint8Array or of length 32.
+ *
+ * @returns {Buffer} DER Public key Prefix + key bytes.
+*/
+function publicKeyDerEncode({publicKeyBytes}) {
+  if(!(publicKeyBytes instanceof Uint8Array && publicKeyBytes.length === 32)) {
+    throw new TypeError('`publicKeyBytes` must be a 32 byte Buffer.');
+  }
+  return Buffer.concat([DER_PUBLIC_KEY_PREFIX, Buffer.from(publicKeyBytes)]);
+}
+
+// seedBytes can be a TypedArray or a function that returns a TypedArray
+const getSeedBytes = async seed => {
+  if(typeof seed === 'function') {
+    return seed();
+  }
+  return seed;
+};
+
+const api = {
+  async generateKeyPairFromSeed(seed) {
+    const seedBytes = await getSeedBytes(seed);
+    const publicBytes = new Uint8Array(32);
+    const privateBytes = new Uint8Array(64);
+    privateBytes.set(seedBytes);
+    privateBytes.set(publicBytes, 32);
+    const privateKey = await createPrivateKey({
+      // node is more than happy to create a new private key using a DER
+      key: privateKeyDerEncode({privateKeyBytes: privateBytes}),
+      format: 'der',
+      type: 'pkcs8'
+    });
+    // this expects either a PEM encoded key or a node privateKeyObject
+    const publicKey = await createPublicKey(privateKey);
+    const publicKeyBuffer = publicKey.export(publicKeyEncoding);
+    const privateKeyBuffer = privateKey.export(privateKeyEncoding);
+    const publicKeyBytes = getKeyMaterial(publicKeyBuffer);
+    const privateKeyBytes = getKeyMaterial(privateKeyBuffer);
+    return {
+      publicKey: publicKeyBytes,
+      secretKey: Buffer.concat([privateKeyBytes, publicKeyBytes])
```

Comment:
We don't need to export the private key at all, right? We can just use the `seedBytes` below?

```suggestion
    const publicKey = getKeyMaterial(publicKeyBuffer);
    return {
      publicKey,
      secretKey: Buffer.concat([seedBytes, publicKeyBytes])
```

Reply:
A few other variables had to be removed in order to remove this the commit is here: https://github.com/digitalbazaar/ed25519-verification-key-2020/pull/3/commits/fa7a27d0bb20a17a023edc036c69f33a4e981229

Reply:
Yay for simplification! :)